lib/include/openamp/virtio.h

```diff
@@ -196,10 +201,133 @@ struct virtio_dispatch {
 	void (*notify)(struct virtqueue *vq);
 };
 
+/**
+ * @brief Create the virtio device virtqueue.
+ *
+ * @param vdev		Pointer to virtio device structure.
+ * @param flags		Create flag.
+ * @param nvqs		The virtqueue number.
+ * @param names		Virtqueue names.
+ * @param callbacks	Virtqueue callback functions.
+ *
+ * @return Pointer to virtio device structure.
+ */
 int virtio_create_virtqueues(struct virtio_device *vdev, unsigned int flags,
 			     unsigned int nvqs, const char *names[],
 			     vq_callback callbacks[]);
 
+/**
+ * @brief Delete the virtio device virtqueue.
+ *
+ * @param vdev	Pointer to virtio device structure.
+ *
+ * @return pointer to virtio device structure.
+ */
+static inline void virtio_delete_virtqueues(struct virtio_device *vdev)
+{
+	return vdev->func->delete_virtqueues(vdev);
```

Comment:
Question, is guaranteed the `func` will be always valid here? It would not be good to check for availabillity of this function before calling, also making this way will obligate the implementor of the dispatch functions to provide this function, is this intended?

Reply:
Yes the point is valid. `Vdev` , 'vdev->func' and 'vdev->func->delete_virtqueues' should be checked.
@CV-Bowen:  It is acceptable for me  to do it in a second step as here the main  objective is to create the API. It can also be done by updating the PR.

Reply:
@arnopo  @uLipe  OK, I will update this PR, another question is should I add this check in all the new virtio_xxx() apis?

Reply:
Done.

Reply:
> Done.

The @uLipe comment is valid for all the `func`. with a return on "-ENXIO" when requestion a return value.

Reply:
@arnopo So I change all the virtio_xxx function return type to int and call dispatch function in these functions?
Like:
```c
static inline int virtio_set_status(struct virtio_device *vdev, uint8_t status)
{
    if (!vdev->func->set_status)
        return -ENXIO;
    vdev->func->set_status(vdev, status);
    return 0;
}
```

Reply:
@CV-Bowen , yes this is the intention, the dispatch functions have a return type, when it is present just return the result of the dispatch functions, taking the example above slightly modified:
```
static inline int virtio_set_status(struct virtio_device *vdev, uint8_t status)
{
    if (!vdev->func->set_status)
        return -ENXIO;
    else 
       return vdev->func->set_status(vdev, status);
}
```
The idea is all the dispatch function templates have an return value.